recipes/theora/all/conanfile.py

```diff
@@ -1,47 +1,26 @@
-from conan.tools.microsoft import msvc_runtime_flag
-from conans import ConanFile, MSBuild, AutoToolsBuildEnvironment, tools
-import functools
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, CMakeDeps, cmake_layout
+from conan.tools.files import copy, download, get
+from conan.tools.microsoft import is_msvc
 import os
-import re
-import shutil
-import stat
-
-required_conan_version = ">=1.36.0"
 
+required_conan_version = ">=1.52.0"
 
 class TheoraConan(ConanFile):
     name = "theora"
-    description = "Theora is a free and open video compression format from the Xiph.org Foundation"
+    license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/xiph/theora"
+    description = "Theora is a free and open video compression format from the Xiph.org Foundation"
     topics = ("theora", "video", "video-compressor", "video-format")
-    license = "BSD-3-Clause"
-
-    settings = "os", "arch", "compiler", "build_type"
-    options = {
-        "shared": [True, False],
-        "fPIC": [True, False],
-    }
-    default_options = {
-        "shared": False,
-        "fPIC": True,
-    }
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
 
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
-
-    @property
-    def _user_info_build(self):
-        return getattr(self, "user_info_build", self.deps_user_info)
+    # original theora.def from: "https://raw.githubusercontent.com/xiph/theora/v1.1.1/lib/theora.def"
+    # edited to change library name to just "theora"
+    exports_sources = "CMakeLists.txt", "conan-theora.def"
```

Comment:
I thought about it, but I didn't like how the old recipe was fetching a secondary file from a place that eventually disappear, AND, this library is basically unmaintained and superseded by newer open-sourced codecs.
So instead I would personally prefer we keep the def file with the recipe and only rely on downloading the source tar ball (which is more likely to be kept around in future webserver rearrangements by gnome).

If there is ever a new release, I would suggest that would be the best time to change the way this works.

I personally only need the library because it is sucked in by some other library.  I don't use theora video codecs.  It was also a nice small library+recipe to try and upgrade to conan v2.